src/sys/socket/mod.rs

```diff
@@ -2076,7 +2079,13 @@ pub fn socket<T: Into<Option<SockProtocol>>>(
 
     let res = unsafe { libc::socket(domain as c_int, ty, protocol) };
 
-    Errno::result(res)
+    match res {
+        -1 => Err(Errno::last()),
+        fd => {
+            // Safe because libc::socket returned success
+            unsafe { Ok(OwnedFd::from_raw_fd(fd)) }
+        }
+    }
```

Comment:
I think this can be more safely and succinctly handled like:
```suggestion
    match res {
        -1 => Err(Errno::last()),
        // Safe because libc::socket returned success
        fd @ 0.. => unsafe { Ok(OwnedFd::from_raw_fd(fd)) },
        _ => unreachable!()
    }
```